### PR TITLE
adding a flag for subscription timeout for check operator cmd

### PR DESCRIPTION
**Pull request by acornett21**
- Adding a flag to set the `subscription timeout` during DeployableByOLM check
- Relates: #1119

### CODE REVIEW

File: internal/policy/operator/deployable_by_olm.go [hunk 1]
```diff
@@ -106,6 +107,14 @@ func WithCSVTimeout(csvTimeout time.Duration) Option {
 	}
 }
 
+// WithSubscriptionTimeout overrides the default subscriptionTimeout, for subscriptions that
+// additional time to install.
```
Comment (komish) [inline internal/policy/operator/deployable_by_olm.go:111]: NIT:

```suggestion
// WithSubscriptionTimeout customizes how long to wait for a subscription to become healthy.
```

Or similar.

File: operator/check_operator.go [hunk 2]
```diff
@@ -191,4 +200,5 @@ type operatorCheck struct {
 	resolved                bool
 	policy                  policy.Policy
 	csvTimeout              time.Duration
+	subscriptionTimeout     time.Duration
```
Comment (komish) [inline operator/check_operator.go:203]: Do you need to default this in NewCheck?

Comment (acornett21) [reply operator/check_operator.go:203]: It gets defaulted in `root`

https://github.com/redhat-openshift-ecosystem/openshift-preflight/pull/1207/files#diff-44de5e8cb8a37326e7eab8c59d8fee1b9cd658555fcfe63af1da9d850cec8fd8R97

File: internal/policy/operator/deployable_by_olm.go (hunk 1 above)
Comment (acornett21) [reply internal/policy/operator/deployable_by_olm.go:111]: This is more readable, I'll edit the CSV language in another PR as well.

File: operator/check_operator.go (hunk 2 above)
Comment (komish) [reply operator/check_operator.go:202]: Viper defaults it for the Preflight CLI, but the library caller isn't going through the same path, right? It's the same reason the [scorecard wait time](https://github.com/redhat-openshift-ecosystem/openshift-preflight/pull/1207/files#diff-6ff614f77f346e6e35530d89670adbcd1ef8b79971019a48350e7ad8065c353fR20) is preconfigured to a default for the library caller.

Comment (acornett21) [reply operator/check_operator.go:203]: I see what you mean now, updated the below, I'll need to go back and add this for CSV as well, in another PR.

https://github.com/redhat-openshift-ecosystem/openshift-preflight/pull/1207/files#diff-6ff614f77f346e6e35530d89670adbcd1ef8b79971019a48350e7ad8065c353fR29

**Review by bcrochet — APPROVED** [review]
/lgtm

**Review by komish — APPROVED** [review]
/lgtm

**Pull request merged**